src/inventory/gilded_rose.js

```diff
@@ -21,48 +21,49 @@ const items = [
 
 updateQuality(items);
 */
+
 export function updateQuality(items) {
-  for (var i = 0; i < items.length; i++) {
-    if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
-      }
+  items.forEach((item) => {
+    const improvesWithAge = item.name === 'Aged Brie' || item.name === 'Backstage passes to a TAFKAL80ETC concert';
+    const constantQuality = item.name === 'Sulfuras, Hand of Ragnaros';
+
+    if (constantQuality) return;
+
+    determineItemQuality(item, improvesWithAge);
+    item.sell_in -= 1;
+  });
+}
+
+// Modifies the quality of an item as time moves on.
+function determineItemQuality(item, improvesWithAge) {
```

Comment:
Code Climate doesn't seem to like the Cognitive Complexity of this function. We could try reducing it by breaking up some of the logic into separate functions.

This code may not work, but something like this:
```js
// Modifies the quality of an item as time moves on.
const isBackstagePass = (item) => {
  if (item.name === "Backstage passes to a TAFKAL80ETC concert") {
    handleBackstagePass(item);
  }
};

const keepItemQualityWithinRange = (item) => {
  // Quality has a max of 50
  if (item.quality > 50) item.quality = 50;

  // Quality cannot be negative
  if (item.quality < 0) item.quality = 0;

  return item.quality;
};

function determineItemQuality(item, improvesWithAge) {
  // Quality degrades twice as fast when sell_in < 0.
  if (item.sell_in < 0) item.quality -= 1;

  // Add quality if the item improves with age.
  if (improvesWithAge) {
    isBackstagePass(item);

    if (item.name !== "Backstage passes to a TAFKAL80ETC concert") {
      item.quality += 1;
    }
  } else {
    // Quality degrades by 1 as default, only if it doesn't improve with age.
    item.quality -= 1;
  }

  item.quality = handleMaxAndNegative(item);
}
```